extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -370,16 +366,21 @@ public function getPage() {
 						$data['nsSpecificContent'] = MercuryApiCategoryHandler::getCategoryContent( $title );
 
 						if ( MercuryApiCategoryHandler::hasArticle( $this->request, $article ) ) {
-							$data['article'] = MercuryApiArticleHandler::getArticleJson( $this->request, $article );
 							$data['details'] = MercuryApiArticleHandler::getArticleDetails( $article );
+							$data['article'] = MercuryApiArticleHandler::getArticleJson( $this->request, $article );
+
+							// Remove namespace prefix from displayTitle, so it can be consistent with title
+							$namespacePrefix = $wgContLang->getNamespaces()[$data['ns']];
+
+							if ( strpos( $data['article']['displayTitle'], $namespacePrefix ) === 0 ) {
+								$data['article']['displayTitle'] = substr( $data['article']['displayTitle'], strlen( $namespacePrefix ) + 1 );
```

Comment:
Can't we provide title without NS prefix to avoid removal? or less pretty create Title object here based on ID and use getText method?